plugins/inputs/fritzbox/fritzbox_test.go

```diff
@@ -0,0 +1,95 @@
+package fritzbox
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tdrn-org/go-tr064/mock"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+func TestConfig(t *testing.T) {
+	// Verify plugin can be loaded from config
+	conf := config.NewConfig()
+	require.NoError(t, conf.LoadConfig("testdata/conf/fritzbox.conf"))
+	require.Len(t, conf.Inputs, 1)
+	f, ok := conf.Inputs[0].Input.(*Fritzbox)
+	require.True(t, ok)
+
+	// Verify successful Init
+	require.NoError(t, f.Init())
+
+	// Verify everything is setup according to config file
+	require.ElementsMatch(t, []string{"http://boxuser:boxpassword@fritz.box:49000/", "http://:repeaterpassword@fritz.repeater:49000/"}, f.URLs)
+	require.Equal(t, []string{"device", "wan", "ppp", "dsl", "wlan", "hosts"}, f.Collect)
+	require.Equal(t, config.Duration(60*time.Second), f.Timeout)
+	require.Equal(t, "secret", f.TLSKeyPwd)
+	require.True(t, f.InsecureSkipVerify)
+}
+
+func TestGather(t *testing.T) {
```

Comment:
Please name this `TestCases` to be consistent with other plugins. This eases reviews and navigating the code.

Reply:
Adapted via a3b25ca0621333f40757fe28df94eda547096b1f